flink-python/pyproject.toml

```diff
@@ -21,7 +21,7 @@ requires = [
     "packaging>=20.5; platform_machine=='arm64'",  # macos M1
     "setuptools>=18.0",
     "wheel",
-    "apache-beam>=2.43.0,<2.49.0",
+    "apache-beam>=2.52.0",
```

Comment:
Should there be an upper bound for dependency version to prevent breaking changes coming in?
Or can we make this version fixed (use `==`)?

Reply:
We made a change previously to make this more permissive 
https://issues.apache.org/jira/browse/FLINK-32758

Reply:
I think we should have visibility into breaking changes on our nightly tests, so it might be better to pull in the latest versions as soon as possible, to mitigate vulnerable code